### PR TITLE
Hann asma

### PR DESCRIPTION
**Pull request by haniasma**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
- *(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)*

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#0000
- *Link to ticket created in va.gov-team repo*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected*
- *Describe the tests completed and the results*
- *Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)*

## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._

| | Before | After |
| --- | --- | --- |
| Mobile | | |
| Desktop | | |

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/secure-messaging/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-custom-folders.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import SecureMessagingSite from '../sm_site/SecureMessagingSite';
```
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-custom-folders.cypress.spec.js:1]: Resolved

File: .gitignore [hunk 2]
```diff
@@ -55,3 +55,4 @@ script/cypress-testrail-helper/my-config.json
 
 # Ignore mochawesome report directory
 mochawesome-report/
+src/applications/mhv/secure-messaging/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-select-dropdown-nav-cypress.spec.js
```
[the remark this answers is not in the document]
Comment (haniasma) [reply .gitignore:58]: I am focusing on other codes I need some more clarification abut this issues please

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      .shadow()
+      .find('button')
+      .contains('Cancel')
+      .realPress(['Enter']);
+  };
+
+  verifyTrash = () => {
+    cy.tabToElement(':nth-child(2) > .usa-button-secondary');
+    cy.realPress(['Enter']);
+    cy.get('[data-testid="delete-message-confirm-note"] p', { timeout: 8000 })
```
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js:57]: Resolved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      .should('be.visible');
+    cy.get('[data-testid="delete-message-modal"]')
+      .shadow()
+      .find('button')
+      .contains('Cancel')
+      .should('be.visible')
+      .realPress(['Enter']);
+  };
+
+  verifyReply = () => {
```
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js:78]: Resolved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  verifyPrintCancelButton = () => {
+    cy.tabToElement(':nth-child(1) > .usa-button-secondary');
+    cy.realPress(['Enter']);
+
+    cy.get('[data-testid="print-modal-popup"]', { timeout: 8000 })
+      .shadow()
+      .find('h1')
+      .contains('What do you want to print?')
+      .should('be.visible');
+    cy.get('[data-testid="radio-print-one-message"]').should('be.visible');
```
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js:40]: Resolve

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
+import mockMessage from '../fixtures/message-response.json';
+import mockThread from '../fixtures/thread-response.json';
+
+class PatientMessageDetailsKeyboardPage {
```
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js:4]: Resolved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientKeywordSearchPage.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
+import mockCategories from '../fixtures/categories-response.json';
+import mockFolders from '../fixtures/folder-response.json';
+import mockInboxFolder from '../fixtures/folder-inbox-response.json';
+import mockMessages from '../fixtures/messages-response.json';
+import mockRecipients from '../fixtures/recipients-response.json';
+
+class PatientKeywordSearchPage {
```
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientKeywordSearchPage.js:7]: Resolved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+      'POST',
+      `/my_health/v1/messaging/message_drafts/${messageId}/replydraft`,
+    ).as('replyDraftSave');
+  };
+
+  verifyPrintCancelButton = () => {
+    cy.tabToElement(':nth-child(1) > .usa-button-secondary');
+    cy.realPress(['Enter']);
+
+    cy.get('[data-testid="print-modal-popup"]', { timeout: 8000 })
```
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js]: Already solved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js]: Already solved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js]: Already solved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js]: Already solved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientMessageDetailsKeyboardPage.js]: Already solved

File: src/applications/mhv/secure-messaging/tests/e2e/pages/PatientKeywordSearchPage.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/pages/PatientKeywordSearchPage.js:7]: Already solved

File: src/applications/mhv/secure-messaging/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-custom-folders.cypress.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (haniasma) [reply src/applications/mhv/secure-messaging/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-custom-folders.cypress.spec.js]: Already solved

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (haniasma) [reply .gitignore:58]: Already solved

**Review by Asterix22 — APPROVED** [review]

**Pull request merged**